numpy/core/tests/test_numeric.py

```diff
@@ -1722,6 +1722,22 @@ def test_clip_with_out_array_outint32(self):
         self.clip(a, m, M, act)
         assert_array_strict_equal(ac, act)
 
+    def test_clip_with_out_transposed(self):
+        # Test that the out argument works when tranposed
+        a = np.arange(16).reshape(4, 4)
+        out = np.empty_like(a).T
```

Comment:
So basically `out` is Fortran contiguous and `clip` now restores C contiguity in `out` so the results don't get assigned memory locations down the columns?

Reply:
Yeah, clip writes to a temporary C-order array. It could also decide to use the "slow" path, which I have no clue if it is much slower when this happens already.